sdk/Managed/src/Microsoft.WindowsAzure.MobileServices/Table/Sync/Queue/OperationQueue.cs

```diff
@@ -149,6 +149,32 @@ public Task<IDisposable> LockItemAsync(string id, CancellationToken cancellation
             }
         }
 
+        public virtual async Task<bool> UpdateAsync(string id, long version, JObject item)
+        {
+            try
+            {
+                MobileServiceTableOperation op = await GetOperationAsync(id);
+                if (op == null || op.Version != version)
+                {
+                    return false;
+                }
+
+                // Change the operation state back to pending since this is a newly updated operation without any conflicts
+                op.State = MobileServiceTableOperationState.Pending;
+                
+                // if the operation type is delete then set the item property in the Operation table
+                if (op.Kind == MobileServiceTableOperationKind.Delete)
+                    op.Item = item;
```

Comment:
else { op.Item = null }, since we aren't enforcing any old type to new type better be safe.